kong/runloop/handler.lua

```diff
@@ -342,6 +348,9 @@ end
 return {
   build_router     = build_router,
   build_api_router = build_api_router,
+  set_check_router_rebuild = function(f)
```

Comment:
Tiny suggestion: in the `balancer` module, we prefixed with a `_` the functions that were exported for unit-testing purposes only (along with a comment); this helps people reading the module to understand that this is not really a "configurable" bit of the module. I can make the change when merging.

Reply:
> the general pattern in Lua probably predates Kong itself

Yeah, I believe it comes from the Python convention of treating `_`-prefixed methods in a class as private-by-convention since they have OOP but no visibility rules

Reply:
> writing this module in the _M form would avoid the need for such a function

but then `check_router_rebuild` would be public. I was sad to even see it escape from the `do` block :) The router logic is delicate, anything to help the reader understand the dependency structure helps!

Reply:
Yes, but testability trumps readability imho (the router rebuild is slightly complicated to the neophyte, but isn’t _too_ cluttered, it is still fairly ok to grasp). And adding tests-only public functions makes readability worse than private functions being public (since the former makes the latter possible anyways, plus adds more code).

Reply:
Sure, don't get me wrong, I don't like test-only functions either. On the readability aspect, I think it's just that whenever I see a public function it adds this extra mental cost of going through "is this used somewhere else? does someone else depends on this?", so I guess I was thinking more about "understandability" than readability.